app/serializers/activity_serializer.rb

```diff
@@ -11,8 +11,59 @@ class ActivitySerializer
              "prov:wasDerivedFrom",
              "prov:wasAttributedTo",
              :action,
-             :version,
-             :changes
+             :version
+
+  attribute :changes do |object, params|
+    changes = object._source.changes
+    pub = changes.publisher
+    pub_obj = changes.publisher_obj
+
+    ret = changes
+
+    if object._source.action == "update"
+      if pub || pub_obj
+        if params[:publisher] == "true"
+          if !pub_obj
+            changes.publisher =
+              [
+                { "name": pub[0] },
+                { "name": pub[1] },
+              ]
+          else
+            changes.publisher = changes.publisher_obj
+          end
+        else
+          if !pub
+            changes.publisher = [ pub_obj[0].name, pub_obj[1].name ]
+          end
+        end
+
+        ret = changes
+      end
+    elsif object._source.action == "create"
+      if pub || pub_obj
+        if params[:publisher] == "true"
+          if !pub_obj
+            changes.publisher = { "name": pub }
+          else
+            changes.publisher = changes.publisher_obj
+          end
+        else
+          if !pub
+            changes.publisher = pub_obj.name
+          end
+        end
+
+        ret = changes
+      end
+    end
+
+    if pub_obj
+      changes.delete("publisher_obj")
+    end
+
+    ret
+  end
```

Comment:
Can we refactor this code like below, I tried to eliminate multiple `if` statements here. 

```
attribute :changes do |object, params|
  changes = object._source.changes
  pub = changes.publisher
  pub_obj = changes.publisher_obj

  if params[:publisher] == "true"
    changes.publisher =
      if pub
        object._source.action == "update" ? [{ "name": pub[0] }, { "name": pub[1] }] : { "name": pub }
      else
        changes.publisher_obj
      end
  elsif pub_obj
    changes.publisher = object._source.action == "update" ? [pub_obj[0].name, pub_obj[1].name] : pub_obj.name
  end

  changes.delete("publisher_obj") if pub_obj

  changes
end
```

Reply:
Done

Reply:
Thank you Suzzane